main.go

```diff
@@ -84,21 +106,29 @@ func newName(vl api.ValueList, index int) string {
 func newLabels(vl api.ValueList, md metadata) prometheus.Labels {
 	labels := prometheus.Labels{}
 
-	if _, ok := md.tags["Application"]; ok {
-		labels["application"] = md.tags["Application"]
+	// Process the expectedTags. At this point of this function call, all the expectedTags should be present
+	// where any missing, expected tag should have already been backfilled with a default value
+	var stackValue, roleValue *string = nil, nil
+	for eTag, _ := range expectedTags {
+		if tagVal, ok := md.tags[eTag]; ok {
+			// Special case for Stack and Role as we need to merge their tag keys/values into a single tag
+			if eTag == ETagStack {
+				stackValue = &tagVal
+			} else if eTag == ETagRole {
+				roleValue = &tagVal
+			} else {
+				labels[strings.ToLower(eTag)] = tagVal
+			}
+		}
 	}
-
-	if _, ok := md.tags["Environment"]; ok {
-		labels["environment"] = md.tags["Environment"]
+	if stackValue != nil && roleValue != nil {
```

Comment:
Should not happen as all the expected tags will be properly populated by backfillTags when newLabels is called. Otherwise, it's also possible that the other tags may not be populated and not just stack and role.